src/main/java/pl/futurecollars/invoice/db/sql/SqlDatabase.java

```diff
@@ -0,0 +1,327 @@
+package pl.futurecollars.invoice.db.sql;
+
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import javax.annotation.PostConstruct;
+import lombok.AllArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.transaction.annotation.Transactional;
+import pl.futurecollars.invoice.db.Database;
+import pl.futurecollars.invoice.model.Car;
+import pl.futurecollars.invoice.model.Company;
+import pl.futurecollars.invoice.model.Invoice;
+import pl.futurecollars.invoice.model.InvoiceEntry;
+import pl.futurecollars.invoice.model.Vat;
+
+@AllArgsConstructor
+public class SqlDatabase implements Database {
+
+  public static final String SELECT_QUERY = "select i.id, i.date, i.number, "
```

Comment:
Nazwa stałej niewiele mówi. Jedynie, że jest to zapytanie typu "SELECT".
Widząc nazwę takiej stałej nie będę wiedział do czego służy, a więc czy powinienem jej użyć. Sytuacja ta zmusza do zagłębiania w się długiego SQLa.

Reply:
poprawiałam